### PR TITLE
Makes game start smoother

### PR DESCRIPTION
**Pull request by Cyberboss**
:cl: Cyberboss
fix: Spawning to the station should now be a less hitchy experience
/:cl:

### CODE REVIEW

**Comment by ChangelingRain** [comment]
Please make very sure that none of these delays let players break anything.

**Comment by Jalleo** [comment]
I do believe there can be a edgecase of joining in after the round is starting to activate which means you dont become a antag/target. Its heavily lag dependant though.

File: code/modules/client/client_procs.dm [hunk 1]
```diff
@@ -156,6 +156,10 @@ var/next_external_rsc = 0
 		preload_rsc = external_rsc_urls[next_external_rsc]
 #endif
 
+	//wait here if the ticker is in the middle of setup
+	while(ticker.current_state == GAME_STATE_SETTING_UP)
+		stoplag()
```
Comment (MrStonedOne) [inline code/modules/client/client_procs.dm]: Jello was referring to joining the round, not joining the server.

How ever there is already protection against attempting to join the round while the ticker is setting up.

Ie, you can remove this commit.

**Comment by MrStonedOne** [comment]
Round start took nearly 60 seconds from when people spawned and could move around to when the "welcome to the station" sound played.

Not sure there is anything bad about that, but it seemed odd.

**Comment by Cyberboss** [comment]
Well, are you sure it was this or one of the other hundred of my PRs that were merged? lol

**Comment by MrStonedOne** [comment]
no idea, but this one seemed most likely =P

**Comment by optimumtact** [comment]
What's the status on this?

**Comment by MrStonedOne** [comment]
last time we tried it, it added a 2 to 4 minute delay on the start of the round. i'll give it another shot on bagil.

**Comment by optimumtact** [comment]
How did it go

**Comment by MrStonedOne** [comment]
it never happened, i was high and got distracted.

**Comment by MrStonedOne** [comment]
even on the new hardware, at 55 players it takes almost 2 minutes for round start to happen.

everybody is walking around, playing, but it some things don't work, joining doesn't work, and then 1:30 to 2:00 minutes later it finishes starting the round.

**Comment by Cyberboss** [comment]
When I'm next available I'm gonna ask you or LeoZ to testmerge it for one round so I can profile the problem

**Comment by Cyberboss** [comment]
@MrStonedOne  Here's the data I got, any ideas? http://pastebin.com/8ZmK4QbN

**Comment by Cyberboss** [comment]
New idea is to just delay the player key transfer until everything is ready, because I believe that all the actions of the players is bogging down the ticker/setup thread

**Comment by Cyberboss** [comment]
Testmerging on sybil, merge if you don't hear any complaints in 24h

**Comment by Cyberboss** [comment]
Nevermind, latejoin = ghosts -_- + no name prefs

**Comment by Cyberboss** [comment]
Yesterday's testing went fine, go ahead and merge this

**Pull request merged**